src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIPacket.cs

```diff
@@ -217,9 +211,8 @@ public int TakeData(byte[] buffer, int dataOffset, int size)
         /// </summary>
         public void Release()
         {
-            _length = 0;
-            _capacity = 0;
             _data = null;
```

Comment:
What is the purpose of Release?  If the object is just going to get dropped, there shouldn't be a need to null out the byte[], and if it's going to get reused, it'd be nice to be able to reuse the same byte[] if it's large enough.

Reply:
That was exactly what I thought. But it turned out the Manual Test https://github.com/dotnet/corefx/issues/27574 were failed due to non-nulled out byte[] after `Release()` method called. `Release()` is now only used for disposing `SNIPacket` object when an error occurred.